src/code/z_vimode.c

```diff
@@ -1,17 +1,214 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/func_800AC030.s")
+void ViMode_LogPrint(OSViMode* viMode) {
+    OSViMode** viMode_ptr = &viMode;
+#define _LOG(line, fmt_str, val)                 \
+    LogUtils_LogThreadId("../z_vimode.c", line); \
```

Comment:
We usually don't define macros inside functions.
Also, we already have a `LOG_ADDRESS` macro you could use.